.github/workflows/extension_ci.yml

```diff
@@ -165,23 +163,46 @@ jobs:
          mv stoml_linux_amd64 stoml
          chmod +x stoml
          sudo mv stoml /usr/local/bin/
+
       - name: Set version strings
         id: versions
         run: |
           echo "SHORT_SHA=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
           echo "TAG_VER=$(/usr/local/bin/stoml Cargo.toml package.version)" >> $GITHUB_OUTPUT
-      - name: Build and upload image
-        run: |
-          docker build --build-arg="PGMQ_VER=${{ steps.versions.outputs.TAG_VER }}" -t pgmq-pg images/pgmq-pg
-          docker tag pgmq-pg quay.io/tembo/pgmq-pg:v${{ steps.versions.outputs.TAG_VER }}
-          docker tag pgmq-pg quay.io/tembo/pgmq-pg:latest
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+  
       - name: Login to Quay
         uses: docker/login-action@v2
         with:
           registry: quay.io/tembo
           username: ${{ secrets.QUAY_USER_TEMBO }}
           password: ${{ secrets.QUAY_PASSWORD_TEMBO }}
-      - name: Push image
-        run: |
-          docker push quay.io/tembo/pgmq-pg:v${{ steps.versions.outputs.TAG_VER }}
-          docker push quay.io/tembo/pgmq-pg:latest
+
+      - name: Build and push -- Commit
+        # push commit build when not a release
+        if: github.event_name != 'release'
+        uses: docker/build-push-action@v5
+        with:
+          file: ./images/pgmq-pg/Dockerfile
+          context: .
+          platforms: linux/amd64, linux/arm64
```

Comment:
maybe only build amd64 on each commit? emulated build takes a long time...

Reply:
Have you considered native ARM runners? I know github actions itself currently doesn't supply those, but third parties do and its not too hard to set up self hosted runners. If the emulated build is taking too long, could be a decent option.

Reply:
yeah that would ideal!